core/src/main/java/org/elasticsearch/index/query/NotQueryParser.java

```diff
@@ -56,12 +55,16 @@ public NotQueryBuilder fromXContent(QueryParseContext parseContext) throws IOExc
                 // skip
             } else if (token == XContentParser.Token.START_OBJECT) {
                 if (parseContext.parseFieldMatcher().match(currentFieldName, QUERY_FIELD)) {
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
                     queryFound = true;
                 } else {
                     queryFound = true;
                     // its the filter, and the name is the field
-                    query = parseContext.parseInnerFilterToQueryBuilder(currentFieldName);
+                    QueryParser queryParser = parseContext.queryParser(currentFieldName);
```

Comment:
Reverted and openes this issue for further discussion: https://github.com/elastic/elasticsearch/issues/13761